apps/antalmanac/src/components/RightPane/SectionTable/SectionTableBody.tsx

```diff
@@ -183,6 +200,100 @@ const InstructorsCell = withStyles(styles)((props: InstructorsCellProps) => {
     return <NoPaddingTableCell className={classes.cell}>{getLinks(instructors)}</NoPaddingTableCell>;
 });
 
+async function getGpaData(deptCode: string, courseNumber: string, instructors: string[]) {
+    const namedInstructors = instructors.filter((instructor) => instructor !== 'STAFF');
+
+    // Get the GPA of the first instructor of this section where data exists
+    for (const instructor of namedInstructors) {
+        const grades = await Grades.queryGrades(
+            deptCode,
+            courseNumber,
+            instructor,
+            useTabStore.getState().activeTab != 1
+        );
+        if (grades?.averageGPA) {
+            return {
+                gpa: grades.averageGPA.toFixed(2).toString(),
+                instructor: instructor,
+            };
+        }
+    }
+
+    return undefined;
+}
+
+interface GPACellProps {
+    deptCode: string;
+    courseNumber: string;
+    instructors: string[];
+}
+
+function GPACell(props: GPACellProps) {
+    const { deptCode, courseNumber, instructors } = props;
+
+    const [gpa, setGpa] = useState('');
+
+    const [instructor, setInstructor] = useState('');
+
+    const [anchorEl, setAnchorEl] = useState<Element>();
+
+    const handleClick = useCallback((event: React.MouseEvent<HTMLElement>) => {
+        setAnchorEl((currentAnchorEl) => (currentAnchorEl ? undefined : event.currentTarget));
+    }, []);
+
+    const hideDistribution = useCallback(() => {
+        setAnchorEl(undefined);
+    }, []);
+
+    useEffect(() => {
+        getGpaData(deptCode, courseNumber, instructors)
+            .then((data) => {
+                if (data) {
+                    setGpa(data.gpa);
+                    setInstructor(data.instructor);
+                }
+            })
+            .catch(console.log);
+    }, [deptCode, courseNumber, instructors]);
+
+    return (
+        <NoPaddingTableCell>
+            <Button
+                style={{
+                    color: isDarkMode() ? 'dodgerblue' : 'blue',
+                    padding: 0,
+                    minWidth: 0,
+                    fontWeight: 400,
+                    fontSize: '1rem',
+                }}
+                onClick={handleClick}
+                variant="text"
+            >
+                {gpa}
```

Comment:
Should this be underlined like how instructors and locations are?

Reply:
I was just thinking about that. Since it doesn't lead away from the site, the location link should not be underlined so it doesn't look as much like a url. We should change that.